net.rocketpool/image/src/app.ts

```diff
@@ -19,145 +23,119 @@ interface Event {
   logIndex: number;
 }
 
+type Provider = ethers.providers.JsonRpcProvider | ethers.providers.WebSocketProvider;
+
 /////////////////////////////////////////////////////////////////////////////////////////
 
 async function readEventsFromETH(
-  startBlock,
-  endBlock,
-  clb: (_: Event) => boolean,
-  stride = 10000
-) {
+  rethContract: RethAbi,
+  startBlock: number,
+  endBlock: number,
+  clb: (_: EventRow) => boolean,
+): Promise<void> {
   console.log(
-    "Considering events in block range [%s, %s] (%s blocks)",
-    startBlock,
-    endBlock,
-    endBlock - startBlock
+    `Considering events in block range [${startBlock}, ${endBlock}] (${endBlock - startBlock})`,
   );
 
-  let fromBlock = startBlock;
-  let toBlock = fromBlock;
-
-  do {
-    fromBlock = toBlock + 1;
-    toBlock = Math.min(endBlock, fromBlock + stride);
+  const eventFilters = [
+    rethContract.filters.TokensMinted(),
+    rethContract.filters.TokensBurned()
+  ] as const;
+  const eventsRowsPromises = eventFilters.map(async (eventFilter) => {
+    const events = await rethContract.queryFilter(eventFilter, startBlock, endBlock);
+
+    return events.map(async (event) => {
+      const block = await rethContract.provider.getBlock(event.blockNumber);
+
+      return {
+        eventName: event.event,
+        amount: event.args.amount.toString(),
+        ethAmount: event.args.ethAmount.toString(),
+        blockNumber: event.blockNumber,
+        blockHash: event.blockHash,
+        blockTime: block.timestamp,
```

Comment:
It's so weird that we get all kinds of info about block/transaction except for the timestamp...

Reply:
Found the related issue: https://github.com/ethers-io/ethers.js/issues/1250

Reply:
The issue blames it on `eth_getLogs` ETH node RPC not returning such data.

Note that the previous code was fetching time from event itself:
```
logDecoded.events.find((i) => i.name == "time").value
```

Reply:
> The issue blames it on eth_getLogs ETH node RPC not returning such data.

Yeah, a lot of web3 backend dev think the same (I'm not an exception):  consider this moment a pain in the ass.

Honesty speaking, I see this moment as the weakest part of the proposed solution but even requesting timestamps it works a lot faster than before

> Note that the previous code was fetching time from event itself:

hm, yes, but... it isn't a block timestamp :thinking:

Reply:
it is `time` event argument

Reply:
Hm, looks like I was confused by the field name (`blockTime`)

If it's intended (using not block timestamp) I just use `time` from events here. It gives us another huge performance boost.

I implement that (to parity with the previous solution) -- and I think we probably need to rename the field from `blockTime` to e.g. `eventTime`.

What do you think?

Reply:
> rename the field from blockTime to e.g. eventTime

Yes, let's do this. I don't think any logic elswhere requires `blockTime`, so we are free to modify the schema.

Reply:
Done